src/execution_plan/ops/op_value_hash_join.c

```diff
@@ -0,0 +1,289 @@
+/*
+* Copyright 2018-2019 Redis Labs Ltd. and Contributors
+*
+* This file is available under the Redis Labs Source Available License Agreement
+*/
+
+#include "op_value_hash_join.h"
+#include "../../value.h"
+#include "../../util/arr.h"
+#include "../../util/qsort.h"
+#include <assert.h>
+
+/* Determins order between two records by inspecting
+ * element stored at postion idx. */
+static bool _record_islt(Record l, Record r, uint idx) {
+    SIValue lv = Record_GetScalar(l, idx);
+    SIValue rv = Record_GetScalar(r, idx);
+    return SIValue_Order(lv, rv) < 0;
+}
+
+/* `idx` is an actual variable in the caller function.
+ * Using it in a macro like this is rather ugly,
+ * but the macro passed to QSORT must accept only 2 arguments. */
+#define RECORD_SORT_ON_ENTRY(a, b) (_record_islt((*a), (*b), idx))
+
+// Performs binary search, returns the leftmost index of a match.
+static int64_t _binarySearchLeftmost(Record *array, int join_key_idx, SIValue v) {    
+    int32_t recordCount = array_len(array);
+    int32_t pos;
+    int32_t left = 0;
+    int32_t right = recordCount;
```

Comment:
All of these variables should be unsigned. Technically, the return should be 32-bit as well, but I don't mind promoting it this way since it may be added to post-return.

Reply:
Changed to int64_t